src/Storage/Repository/PgApplicationRepository.cs

```diff
@@ -0,0 +1,198 @@
+                await using NpgsqlDataReader reader = await pgcom.ExecuteReaderAsync();
+                if (await reader.ReadAsync())
+                {
+                    application = SetLegacyId(reader.GetFieldValue<Application>("application"));
+                    _memoryCache.Set(appId, application, _cacheEntryOptionsMetadata);
+                }
+                else
+                {
+                    application = null;
+                }
+            }
+
+            return application;
+        }
+
+        /// <inheritdoc/>
+        public async Task<Application> Create(Application item)
+        {
+            SetInternalId(item);
+            await using NpgsqlCommand pgcom = _dataSource.CreateCommand(_createSql);
+            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, item.Id);
+            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, item.Org);
+            pgcom.Parameters.AddWithValue(NpgsqlDbType.Jsonb, item);
+
+            await pgcom.ExecuteNonQueryAsync();
+
+            return SetLegacyId(item);
+        }
+
+        /// <inheritdoc/>
+        public async Task<Application> Update(Application item)
+        {
+            SetInternalId(item);
+            await using NpgsqlCommand pgcom = _dataSource.CreateCommand(_updateSql);
+            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, item.Id);
+            pgcom.Parameters.AddWithValue(NpgsqlDbType.Jsonb, item);
+            await pgcom.ExecuteNonQueryAsync();
+
+            SetLegacyId(item);
+            _memoryCache.Set(item.Id, item, _cacheEntryOptionsMetadata);
+            return item;
+        }
+
+        /// <inheritdoc/>
+        public async Task<bool> Delete(string appId, string org)
+        {
+            await using NpgsqlCommand pgcom = _dataSource.CreateCommand(_deleteSql);
+            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, appId.Replace('/', '-'));
+
+            return await pgcom.ExecuteNonQueryAsync() == 1;
+        }
+
+        /// <inheritdoc/>
+        public async Task<Dictionary<string, string>> GetAllAppTitles()
+        {
+            if (!_memoryCache.TryGetValue(_cacheKey, out Dictionary<string, string> appTitles))
+            {
+                appTitles = new Dictionary<string, string>();
+                foreach (Application item in await FindAll())
+                {
+                    StringBuilder titles = new();
+                    if (item?.Title?.Values != null)
+                    {
+                        foreach (string title in item.Title.Values)
+                        {
+                            titles.Append(title + ";");
+                        }
+                    }
+
+                    appTitles.Add(item.Id, titles.ToString());
+                }
+
+                _memoryCache.Set(_cacheKey, appTitles, _cacheEntryOptionsTitles);
+            }
+
+            return appTitles;
+        }
+
+        private static Application SetLegacyId(Application app)
```

Comment:
Now that we no longer need to avoid / in the id because of limitations in CosmosDB, we could choose to store the value with / and avoid the headache of maintaining this special case.

Reply:
Such a change will also influence the id in the text resource. It's not recommended to do this now.